lib/route53.js

```diff
@@ -28,53 +74,23 @@ class AWSHandler extends DNSHandler {
 		return this._awsSdk;
 	}
 
-	_getHostedZoneId() {
+	getHostedZoneId() {
 		return this.Route53.listHostedZonesByName({
 			DNSName: this.domain,
 		}).promise()
-			.then(response => _.filter(_.flatMap(response), [
-				"Config.PrivateZone",
-				false,
-			])[0].Id)
-			.catch(error => {
-				console.log("[ERROR]", error);
-			});
-	}
-
-	_generateParams(hostedZoneId, opts) {
-		let params = {
-			HostedZoneId: hostedZoneId,
-			ChangeBatch: {
-				Changes: [
-					{
-						Action: "",
-						ResourceRecordSet: {
-							Name: opts.name.concat(".", this.domain, "."),
-							Type: opts.type || "CNAME",
-							TTL: opts.ttl || 3600,
-						},
-					},
-				],
-			},
-		};
-
-		if (opts.values) {
-			params.ChangeBatch.Changes[0].ResourceRecordSet.ResourceRecords = opts.values.map(function(r){
-				return {
-					Value: r
-				}
-			});
-		}
-		return Promise.resolve(params);
+		.then(response => _.filter(_.flatMap(response), [
+			"Config.PrivateZone",
+			false,
+		])[0].Id)
```

Comment:
This might be dangerous as you may not have gotten any results, in which case `[0]` will blow up in spectacular ways.